api/src/main/java/org/openmrs/api/OrderExceptions.java

```diff
@@ -0,0 +1,170 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.api;
+
+import static org.openmrs.Order.Action.DISCONTINUE;
+
+/**
+ * Represents specific error conditions that are part of OrderService.
+ * The specific error classes define only a default constructor if they are meant to be used with only a single error message.
+ * In other cases, there should be static factory methods that can be used to create the error class with the specific error message needed.
+ */
+public class OrderExceptions {
+
+	public static final long serialVersionUID = 22121212L;
+	
+	/**
+	 * Thrown when the OrderAlreadyExists.
+	 */
+	public static class CannotEditAlreadyExistingOrderException extends APIException {
+
+		public static final long serialVersionUID = 22121213L;
+		
+		public CannotEditAlreadyExistingOrderException() {
+			super("Order.cannot.edit.existing");
+		}
+	}
+
+	/**
+	 * Thrown when the Order has already been discontinued.
+	 */
+	public static class CannotDiscontinueAlreadyDiscontinuedOrderException extends APIException {
+
+		public static final long serialVersionUID = 22121214L;
+		
+		public CannotDiscontinueAlreadyDiscontinuedOrderException() {
+			super("Order.stopped.cannot.discontinued");
+		}
+	}
+	
+	/**
+	 * Thrown when trying to stop a retrospective discontinued order.
+	 */
+	public static class CannotStopRetrospectiveDiscontinuedOrderException extends APIException {
+		
+		public static final long serialVersionUID = 22221215L;
+		
+		public CannotStopRetrospectiveDiscontinuedOrderException() {
+			super("Order.retrospective.stopped.cannot.discontinued");
+		}
+	}
+
+	/**
+	 * Thrown with trying to discontinue an order with a specific action that cannot be discontinued.
+	 */
+	public static class CannotDiscontinueOrderWithActionException extends APIException {
+
+		public static final long serialVersionUID = 22121215L;
+		
+		public CannotDiscontinueOrderWithActionException() {
+			super("Order.action.cannot.discontinued", new Object[] { DISCONTINUE }, null);
```

Comment:
If just passing null for the third argument, why not just use the two argument constructor? :)